lib/controllers/items.js

```diff
@@ -1,5 +1,39 @@
 const { Router } = require('express');
+const authorizeItem = require('../middleware/authorizeItem');
+const Item = require('../models/Item');
 
-module.exports = Router();
+module.exports = Router()
+  .delete('/:id', authorizeItem, async (req, res, next) => {
+    try {
+      const resp = await Item.delete(req.params.id);
+      res.json(resp);
+    } catch (e) {
+      next(e);
+    }
+  })
+  .put('/:id', authorizeItem, async (req, res, next) => {
+    try {
+      const item = await Item.updateById(req.params.id, req.body);
+      res.json(item);
+    } catch (e) {
+      next(e);
+    }
+  })
+  .post('/', async (req, res, next) => {
+    try {
+      const item = await Item.insert({ ...req.body, user_id: req.user.id });
+      res.json(item);
+    } catch (e) {
+      next(e);
+    }
+  })
+  .get('/', async (req, res, next) => {
+    try {
+      const resp = await Item.getAll(req.user.id);
+      res.json(resp);
+    } catch (e) {
+      next(e);
+    }
```

Comment:
Don't forget to add in the authenticate middleware to all of our routes so that non-authenticated users cannot make changes to our data.